std/exception.d

```diff
@@ -2145,3 +2145,58 @@ pure nothrow @safe unittest
 
     auto infSlice = infinite[0 .. $]; // this would throw otherwise
 }
+
+
+/**
+ * Convenience mixin for trivially sub-classing exceptions
+ *
+ * Even trivially sub-classing an exception involves writing boilerplate code
+ * for the constructor to correctly pass in the source file and line number the
+ * exception was thrown from. This mixin makes it easier.
+ *
+ * Note that if you want to pass in special arguments to your exception, you
+ * obviously can't use this mixin and have to write the constructor manually.
+ */
+mixin template StandardExceptionConstructors()
```

Comment:
Re the history,  I had by error branched off from my regex branch i.o. `master`. To clean it up I just did a painless `git rebase --onto master <wrong_branch> <current_commit>`, deleted the branch name and checked out the detached state as the same branch name and pushed.

I'm a humanities researcher and programmer by interest, not profession. :-) So thanks for putting up with such bloopers on my part!